### PR TITLE
Javadocs #145

### PR DESCRIPTION
**Pull request by prb112**
### Commands
```
export JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/
mvn site javadoc:aggregate -f fhir-parent/pom.xml -DskipTests  -X
```

### RESULTS 
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for IBM FHIR Server 4.0.0-SNAPSHOT:
[INFO]
[INFO] IBM FHIR Server .................................... SUCCESS [28:11 min]
[INFO] fhir-core .......................................... SUCCESS [ 21.551 s]
[INFO] fhir-database-utils ................................ SUCCESS [01:23 min]
[INFO] fhir-config ........................................ SUCCESS [ 21.890 s]
[INFO] fhir-audit ......................................... SUCCESS [ 23.351 s]
[INFO] fhir-model ......................................... SUCCESS [22:56 min]
[INFO] fhir-registry ...................................... SUCCESS [ 22.462 s]
[INFO] fhir-validation .................................... SUCCESS [ 21.657 s]
[INFO] fhir-ig-us-core .................................... SUCCESS [ 17.939 s]
[INFO] fhir-search ........................................ SUCCESS [ 31.718 s]
[INFO] fhir-persistence ................................... SUCCESS [ 34.669 s]
[INFO] fhir-persistence-proxy ............................. SUCCESS [ 22.904 s]
[INFO] fhir-persistence-schema ............................ SUCCESS [ 23.560 s]
[INFO] fhir-persistence-jdbc .............................. SUCCESS [ 37.040 s]
[INFO] fhir-provider ...................................... SUCCESS [ 19.424 s]
[INFO] fhir-operation ..................................... SUCCESS [ 22.260 s]
[INFO] fhir-operation-validate ............................ SUCCESS [ 13.641 s]
[INFO] fhir-operation-document ............................ SUCCESS [ 13.310 s]
[INFO] fhir-operation-healthcheck ......................... SUCCESS [ 13.223 s]
[INFO] fhir-operation-apply ............................... SUCCESS [ 13.408 s]
[INFO] fhir-client ........................................ SUCCESS [ 24.371 s]
[INFO] fhir-operation-bulkdata ............................ SUCCESS [ 25.166 s]
[INFO] fhir-bulkimportexport-webapp ....................... SUCCESS [ 16.205 s]
[INFO] fhir-cli ........................................... SUCCESS [ 17.001 s]
[INFO] fhir-notification .................................. SUCCESS [ 15.243 s]
[INFO] fhir-notification-kafka ............................ SUCCESS [ 14.042 s]
[INFO] fhir-notification-websocket ........................ SUCCESS [ 14.441 s]
[INFO] IBM FHIR Server - REST API Implementation .......... SUCCESS [ 26.054 s]
[INFO] IBM FHIR Server - REST API WebApp .................. SUCCESS [ 10.123 s]
[INFO] fhir-server-test ................................... SUCCESS [ 25.213 s]
[INFO] fhir-swagger-generator ............................. SUCCESS [ 14.571 s]
[INFO] fhir-openapi ....................................... SUCCESS [ 10.002 s]
[INFO] fhir-install ....................................... SUCCESS [  9.495 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:31 h
[INFO] Finished at: 2019-11-13T12:21:06-05:00
[INFO] ------------------------------------------------------------------------
```

### Command
`mvn -Pfhir-javadocs site -f fhir-examples/pom.xml`

### Results 
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.089 s
[INFO] Finished at: 2019-11-13T12:27:58-05:00
[INFO] ------------------------------------------------------------------------
```

### Command 
`mvn -Pfhir-javadocs site -f fhir-tools/pom.xml`

### Results
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  30.398 s
[INFO] Finished at: 2019-11-13T12:31:51-05:00
[INFO] ------------------------------------------------------------------------
```

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-examples/pom.xml [hunk 1]
```diff
@@ -40,63 +40,63 @@
...
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-project-info-reports-plugin</artifactId>
+                        <version>2.9</version>
+                    </plugin>
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-javadoc-plugin</artifactId>
-                        <version>3.1.1</version>
+                        <version>3.0.1</version>
```
Comment (lmsurpre) [inline fhir-examples/pom.xml]: why the downgrade?

File: fhir-tools/pom.xml [hunk 2]
```diff
@@ -290,63 +290,63 @@
...
+                            </additionalparam>
+                            <!-- skips the three entries -->
+                            <nohelp>true</nohelp>
+                            <nodeprecated>true</nodeprecated>
+                            <nodeprecatedlist>true</nodeprecatedlist>
                         </configuration>
                         <executions>
                             <execution>
-                                <id>aggregate</id>
+                                <id>attach-javadocs</id>
```
Comment (lmsurpre) [inline fhir-tools/pom.xml]: here its `attach-javadocs` but in the PR description the commands are like `mvn site javadoc:aggregate -f fhir-parent/pom.xml -DskipTests  -X` ...just making sure its right

Comment (lmsurpre) [reply fhir-tools/pom.xml]: oh, i see the `<phase>site</phase>` now...I guess the `mvn -Pfhir-javadocs site -f fhir-examples/pom.xml` will invoke this execution now.

Comment (prb112) [reply fhir-tools/pom.xml]: correct. that way it's not part of the main build.

File: fhir-examples/pom.xml (hunk 1 above)
Comment (prb112) [reply fhir-examples/pom.xml]: good catch - thanks... it was part of my debugging process... which was extensive :)

File: fhir-tools/pom.xml [hunk 3]
```diff
@@ -290,63 +290,63 @@
...
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-project-info-reports-plugin</artifactId>
+                        <version>2.9</version>
+                    </plugin>
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-javadoc-plugin</artifactId>
-                        <version>3.1.1</version>
+                        <version>3.0.1</version>
```
Comment (lmsurpre) [inline fhir-tools/pom.xml]: revert this one too i guess

File: fhir-parent/pom.xml [hunk 4]
```diff
@@ -500,6 +500,66 @@
...
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-project-info-reports-plugin</artifactId>
+                    <version>2.9</version>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-javadoc-plugin</artifactId>
+                    <version>3.0.1</version>
```
Comment (lmsurpre) [inline fhir-parent/pom.xml]: use version 3.1.1 to match others

**Pull request merged**